pkg/reconciler/dependencybuild/dependencybuild.go

```diff
@@ -756,7 +759,7 @@ func (r *ReconcileDependencyBuild) handleBuildPipelineRunReceived(ctx context.Co
 		}
 
 	}
-	return reconcile.Result{}, nil
+	return artifactbuild.RemovePipelineFinalizer(ctx, pr, r.client)
```

Comment:
This has always bugged me about github PR reviews, that it only allows comments around changed lines :-) 

But if you go up 12 lines @stuartwdouglas to line 750 here, where you `return r.handleStateCompleted(ctx, &db, log)` 

Looking at that method, I don't see any removing of the finalizer there.

So it looks like that might be a path where you leak the finalizer, if you don't get any more PipelineRun Events after whichever one lead you down that path.

Of course, it has been a while since I've looked at this code, so I might be missing something

Reply:
I believe you want to follow the pattern you used in artifactbuild.go:

```
	err = r.client.Status().Update(ctx, &abr)
	if err != nil {
		return reconcile.Result{}, err
	}
	result, err2 := RemovePipelineFinalizer(ctx, pr, r.client)
	if err2 != nil {
		return result, err2
	}
```

where you updated the ab, returned if you got an error, but otherwise went on the remove the finalizer

Reply:
Good catch, I missed that return statement.